baybe/simulation/lookup.py

```diff
@@ -0,0 +1,209 @@
+"""Target lookup mechanisms."""
+
+import warnings
+from typing import Callable, Collection, List, Literal, Union
+
+import numpy as np
+import pandas as pd
+
+from baybe.targets import NumericalTarget, TargetMode
+from baybe.targets.base import Target
+from baybe.utils import add_fake_results
+
+CallableLookup = Callable[[pd.DataFrame], pd.DataFrame]
```

Comment:
Just to make sure: Did you verify that this is rendered properly in the documentation?

Reply:
Good that you mention it. Yes, it looks good, but shouldn't actually be shown in the doc, and also it didn't follow the naming convention for type aliases. Changed it to `_TCallableLookup` 👍🏼 Resolve if OK

Reply:
Just out of curiosity: Why the `T`?